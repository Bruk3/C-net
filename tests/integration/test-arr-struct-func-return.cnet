int _EXRATE = 6;
int _RICHTHRESHOLD = 100000;

struct employee {
	int pid;
    int salary;
	string manager;
};

int convert_cedi(int b) {return b * _EXRATE;}
int check_rich(struct employee e) {return e.salary >= _RICHTHRESHOLD;}

int main() {
    // Print array elements 
    int[] x = new int[3]{1,2,3};
    int i = 0;

    for(i; i < 3; i = i + 1) {
        stdout.writeln(soi(x[i]) + "");
    }

    // Print struct members 
    struct employee e = new struct employee; 
    e.pid = 1; 
    e.salary = 100000; 
    e.manager = "Bob"; 
    stdout.writeln("Employee " + soi(e.pid) + " under " + e.manager + " earns " + soi(e.salary) + " USD"); 


    // Get conversion from convert_cedi()
    int sal_cedi = convert_cedi(e.salary); 
    stdout.writeln("Employee " + soi(e.pid) + " under " + e.manager + " earns " + soi(sal_cedi) + " CEDI");

    // Check if wealthy
    if (check_rich(e)) {stdout.writeln("Employee " + soi(e.pid) + " is rich");}
    else {stdout.writeln("Employee " + soi(e.pid) + " needs a raise");}

    return 0;
}