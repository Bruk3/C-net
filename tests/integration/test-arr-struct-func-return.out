1
2
3
Employee 1 under Bob earns 100000 USD
Employee 1 under Bob earns 600000 CEDI
Employee 1 is rich
